src/UI/Http/Rest/Controller/User/SignUpController.php

```diff
@@ -41,26 +38,24 @@
      *     name="user",
      *     type="object",
      *     in="body",
-     *     required=true,
      *     schema=@SWG\Schema(type="object",
      *         @SWG\Property(property="uuid", type="string"),
-     *         @SWG\Property(property="email", type="string")
+     *         @SWG\Property(property="email", type="string"),
+     *         @SWG\Property(property="password", type="string")
      *     )
      * )
      *
      * @SWG\Tag(name="User")
      *
-     * @Security(name="Bearer")
-     *
      * @throws \Assert\AssertionFailedException
      */
     public function __invoke(Request $request): JsonResponse
     {
-        $uuid = $request->get('uuid');
+        $uuid = $request->get('uuid') ?: Uuid::uuid4()->toString();
```

Comment:
If we delegate to the Api the uuid generation but we return a 201... How can the client know ID of the resource created?

Reply:
Hm, for call `/api/auth_check` client don't need uuid, after login client can call `/user/{email}`

Reply:
And also I can return uuid :)

Reply:
What is your opinion?

Reply:
I'm planing to replace the `GET /user/{email}` endpoint for `POST /user/_search` as I don't like the email in the url.

My approach... delegate the uuid creation to the client, it simplify all the things

Reply:
Done 👍 
And i'll think about using this approach in project...

Reply:
When working with rest and hypermedia, you can also return a 201 and add a location header with the route of the new resource